### DISCUSSION TITLE
@entangle doesn't work correct, when I use global state in Alpine.js

### DISCUSSION THREAD

**Discussion by Dakasha26** [livewire/livewire#5514] [1. Help] [answered]
Hello!

I'm trying to share data with Alpine.js between several Livewire-components. Then I use global state like:

` $store.global.resourceName`

It works when I use `$wire`, but doesnt work with `@entangle`

My x-data in nested component:

`$store.global.resourceName: @entangle('resourceName').defer`

This variant of code works as I want:

`<button @click="
    $wire.set('resourceName', $store.global.resourceName');
    $wire.call('foo');
">Set var</button>`

But this send `null` to the component:

`<button @click="
    $wire.call('foo');
">Set var</button>`

Thanks for your help!

**Comment by joshhanley** [comment, marked as answer]
@Dakasha26 yeah that's correct, `@entangle` can't be used with global state at this stage. So you will need to `@entangle` to a local property in your Alpine component and then watch that value to sync it manually to the store.

Hope this helps!

**Reply by Dakasha26** [reply]
Thank you!

**Reply by joshhanley** [reply]
No worries! 🙂
